src/librustc_trait_selection/traits/select.rs

```diff
@@ -502,9 +503,43 @@ impl<'cx, 'tcx> SelectionContext<'cx, 'tcx> {
                     None,
                 ) {
                     Ok(_) => Ok(EvaluatedToOk),
+                    Err(ErrorHandled::TooGeneric) => Ok(EvaluatedToAmbig),
                     Err(_) => Ok(EvaluatedToErr),
                 }
             }
+
+            ty::Predicate::ConstEquate(c1, c2) => {
+                debug!("evaluate_predicate_recursively: equating consts c1={:?} c2={:?}", c1, c2);
```

Comment:
Hmm we should try to factor a helper to share the code between this and fulfill.rs

Reply:
I tried and failed. :sweat_smile: the `evaluate` closures are different enough that my approaches at extracting part of them into a shared function ended up making the code worse :/